src/app/components/student-list/student-list.component.ts

```diff
@@ -67,10 +70,17 @@ export class StudentListComponent implements CourseConfigurable {
             await this.refreshStudentList(this.studentSearchTerm);
         }
     }
-    //Configure the course information
+
+    // Configure the course information
     async configureCourse(course: Course): Promise<void> {
         this.course = course;
         this.configuration = await this.tokenATMConfigurationManagerService.getTokenATMConfiguration(this.course);
+        this.isLogPublished = await checkAndConfirmTokenATMLogPublished(
```

Comment:
Unlike request processing, which checks the publish state of Token ATM upon action, the Students page caches the publish state here. It means that if the user choose to publish the Token ATM Log by themselves, the Students page won't refresh the publish state until next time the user visit it. It is recommended to add a line of warning at the top of the page to let the user know that they need to refresh the page (by navigating to another page and navigating back).